src/container/RelayRenderer.js

```diff
@@ -253,29 +212,7 @@ class RelayRenderer extends React.Component {
    * @private
    */
   _shouldUpdate(): boolean {
-    const {activeContainer, activeEnvironment, activeQueryConfig} = this.state;
-    const {Container, queryConfig, environment} = this.props;
-    return (
-      (!activeContainer || Container === activeContainer) &&
-      (!activeEnvironment || environment === activeEnvironment) &&
-      (!activeQueryConfig || queryConfig === activeQueryConfig)
-    );
-  }
-
-  /**
-   * @private
-   */
-  _runQueriesAndSetState(props: RelayRendererProps): void {
-    this._runQueries(props);
-    this.setState(
-      this._buildState(
-        this.state.activeContainer,
-        this.state.activeEnvironment,
-        this.state.activeQueryConfig,
-        null,
-        null
-      )
-    );
+    return !!this.state.readyState || !this.state.active;
```

Comment:
Oooh, got it. To cement my understanding...

- `active` will only be `false` in between the initial render and the first "tick" (when `onReadyStateChange` first gets fired).
- `readyState` will only be non-null if the `onReadyStateChange` callback has ever fired for the current set of `(container, environment, route)`.

Did I get that right?